src/main/java/command/DeleteCommand.java

```diff
@@ -0,0 +1,62 @@
+package command;
+import task.*;
+import duke.*;
+
+/**
+ * DeleteCommand represents a delete command.
+ *
+ * @author Ho Wen Zhong
+ */
+public class DeleteCommand extends Command {
+
+    private int deleteIndex;
+
+    public DeleteCommand(int deleteIndex) {
+        this.deleteIndex = deleteIndex;
+    }
+
+    /**
+     * Deletes task from TaskList, calls showResponse with response,
+     * saves TaskList to file.
+     *
+     * @param tasks Current TaskList.
+     * @param ui Ui object.
+     * @param storage Storage object.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        Task task = tasks.get(deleteIndex);
+        tasks.delete(deleteIndex);
```

Comment:
If the index is negative or too big, it will throw an IndexOutOfBoundsException. Should probably check for it.